packages/astro/src/runtime/server/render/astro/render.ts

```diff
@@ -165,3 +166,101 @@ async function bufferHeadContent(result: SSRResult) {
 		}
 	}
 }
+
+export async function renderToAsyncIterable(
+	result: SSRResult,
+	componentFactory: AstroComponentFactory,
+	props: any,
+	children: any,
+	isPage = false,
+	route?: RouteData
+): Promise<AsyncIterable<Uint8Array> | Response> {
+  const templateResult = await callComponentAsTemplateResultOrResponse(
+    result,
+    componentFactory,
+    props,
+    children,
+    route
+  );
+  if (templateResult instanceof Response)
+    return templateResult;
+  let renderedFirstPageChunk = false;
+  if (isPage) {
+    await bufferHeadContent(result);
+  }
+
+	let error: Error | null = null;
+  let next = promiseWithResolvers<void>();
+  const chunks: Uint8Array[] = []; // []Uint8Array
+
+  const iterator = {
+    async next() {
+      await next.promise;
+
+			if(error) {
+				throw error;
+			}
+
+      // Get the total length of all arrays.
+      let length = 0;
+      chunks.forEach(item => {
+        length += item.length;
+      });
+
+      // Create a new array with total length and merge all source arrays.
+      let mergedArray = new Uint8Array(length);
+      let offset = 0;
+      chunks.forEach(item => {
```

Comment:
Here we can use a `for..of` loop

Reply:
I want to avoid for/of since this is perf critical code. I can change to a tranditional for loop if you prefer though.

Reply:
Switched to a regular for loop.